### PR TITLE
feature: support `luarocks` mode for installation and deployment.

### LINKED ISSUE #300: feature: support `luarocks` mode for installation and deployment.
Because different operating system distributions have different dependencies and installation steps, in order to avoid installation problems caused by system dependencies, we will provide `luarocks` as the new default installation method.
Support `Orange` version: v0.8.0

由于不同的操作系统发行版本有不同的依赖和安装步骤，为了避免系统依赖导致的安装问题，我们将提供`luarocks`作为新的默认安装方式。
支持 `Orange` 版本：v0.8.0

### PR DESCRIPTION
**Pull request by shuaijinchao**
This feature has been tested in the following servers in Alibaba Cloud.
```bash
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.2.1511 (Core)
Release:	7.2.1511
Codename:	Core
```
```bash
LSB Version:	core-9.20170808ubuntu1-noarch:security-9.20170808ubuntu1-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
```
Server Model: ecs.t6-c4m1.large (2 vCPU 512 MB)
#### Review
Because the code is not officially merged into the `master`, please use the `Install of Development Environment` mode to perform the installation test on this machine.
#### Issues
Fix #300 
#### Thanks
The idea and part of the code for this feature comes from APISIX, thanks to the APISIX team.

### CODE REVIEW

File: install/install-orange.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+LUA_ROCKS_VER=`luarocks --version | grep -Eo  "luarocks [0-9]+"`
+echo $LUA_ROCKS_VER
+
+
+UNAME=`uname`
+echo $UNAME
+
+
+do_install() {
+    if [ "$UNAME" == "Darwin" ]; then
```
Comment (membphis) [inline install/install-orange.sh]: do we need to support macOS?

Comment (shuaijinchao) [reply install/install-orange.sh]: current not supported, I am correcting it.

File: Makefile [hunk 2]
```diff
@@ -23,9 +23,9 @@ help:
 .PHONY: dev
 dev:
 ifneq ($(LUAROCKS_VER),'luarocks 3')
-	luarocks install rockspec/orange-0.8-0.rockspec --tree=deps --only-deps --local
+	luarocks install rockspec/orange-dev-1.0-0.rockspec --tree=deps --only-deps --local
```
Comment (membphis) [inline Makefile]: orange-dev-1.0 -> orange-master-1.0

how about this name? ^_^

Comment (shuaijinchao) [reply Makefile]: very good, this is more reasonable.

**Comment by zhangbao0325** [comment]
本地环境 Ubuntu 18.04.3 LTS，新机器按照READNE.md文档顺利安装成功。
如果本地已经源码安装了OR，则安装需环境依赖时，将以下命令中的 "oprenresty openresty-resty" 去除就行。
sudo apt-get install -y openresty openresty-resty curl git luarocks\
    check libpcre3 libpcre3-dev libjemalloc-dev \
    libjemalloc1 build-essential libtool libssl1.0-dev automake autoconf pkg-config cmake

File: Makefile [hunk 3]
```diff
@@ -1,46 +1,60 @@
...
+INST_PREFIX ?= /usr
+INST_LUADIR ?= $(INST_PREFIX)/share/lua/5.1
+INST_BINDIR ?= /usr/bin
+INSTALL ?= install
+COPY ?= cp
+LINK ?= ln
+REMOVE ?= rm
+CHMOD ?= chmod
+OR_EXEC ?= $(shell which openresty)
+LUA_JIT_DIR ?= $(shell ${OR_EXEC} -V 2>&1 | grep prefix | grep -Eo 'prefix=(.*?)/nginx' | grep -Eo '/.*/')luajit
```
Comment (membphis) [inline Makefile]: I think we can drop OR_EXEC and LUA_JIT_DIR, it is useless.

Comment (shuaijinchao) [reply Makefile]: OK, I have deleted it.

**Comment by membphis** [comment]
I think we should make a installation test: lua(5.3 or lua 5.2) + orange.

Standard Lua + OpenResty , they are very different.

File: README.md [hunk 4]
```diff
@@ -8,69 +8,95 @@
...
+cd lor
+sudo make install
 ```
 
-2) Generate configuration file
+#### 3) Install Orange
+
+```bash
+curl -Lo install.sh https://raw.githubusercontent.com/orlabs/orange/master/install/install-orange.sh
+sudo sh install.sh
```
[the remark this answers is not in the document]
Comment (shuaijinchao) [reply README.md]: 1. please see the `review` description in Pr. Before the pr is not officially merged, you need to install it using the `Install of Development Environment` method.
2. this installation method was developed for the `v0.8` version.

**Comment by shuaijinchao** [comment]
> I think we should make a installation test: lua(5.3 or lua 5.2) + orange.
> 
> Standard Lua + OpenResty , they are very different.

OK, I will test it separately and explain it in the documentation.
Thanks.

**Comment by wujunze** [comment]
```
[deploy@code-2001 ~]$ uname -a
Linux code-2001 3.10.0-693.2.2.el7.x86_64 #1 SMP Tue Sep 12 22:26:13 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[deploy@code-2001 ~]$ cat /etc/redhat-release
CentOS Linux release 7.4.1708 (Core)
[deploy@code-2001 ~]$

```

安装测试通过

**Pull request merged**